examples/threading/threading.c

```diff
@@ -28,6 +58,21 @@ bool start_thread_obtaining_mutex(pthread_t *thread, pthread_mutex_t *mutex,int
      *
      * See implementation details in threading.h file comment block
      */
+     
+     int ret;
+     struct thread_data* thread_args = (struct thread_data *) malloc(sizeof(struct thread_data));
+     thread_args->wait_to_obtain_ms = wait_to_obtain_ms;
+     thread_args->wait_to_release_ms = wait_to_release_ms;
+     thread_args->mutex = mutex;
+     
+     ret = pthread_create(thread, NULL, threadfunc, thread_args);
+     if(!ret){
+         errno = ret;
+         perror("thread create");
+         DEBUG_LOG("thread create success\n");
+         return true;
+     }
+    ERROR_LOG("thread create failed\n");
```

Comment:
@TejaswiniLakshminarayan what happens to `thread_args` when `pthread_create` fails?